src/cmd4.cc

```diff
@@ -181,7 +181,7 @@ void do_cmd_change_name(void)
 		else if (c == 'f')
 		{
 			strnfmt(tmp, 160, "%s.txt", player_name);
-			if (get_string("Filename(you can post it to http://angband.oook.cz/): ", tmp, 80))
+			if (get_filename("Filename(you can post it to http://angband.oook.cz/): ", tmp, 80))
```

Comment:
Honestly, I think trying to made this code base secure is probably a fool's errand. I think the best bet would probably to have a sandbox of some type -- or similar.